projects/Gibbed.MassEffect2.FileFormats/SaveFileBase.cs

```diff
@@ -0,0 +1,136 @@
+        public float SecondsPlayed;
+
+        [UnrealFieldCategory("5. Other")]
+        [UnrealFieldDisplayName("Disc")]
+        public int Disc;
+
+        [UnrealFieldCategory("3. Location")]
+        [UnrealFieldDisplayName("Base Level Name")]
+        public string BaseLevelName;
+
+        [UnrealFieldCategory("1. Information")]
+        [UnrealFieldDisplayName("Difficulty")]
+        public Save.DifficultyOptions Difficulty;
+
+        [UnrealFieldCategory("4. Plot")]
+        [UnrealFieldDisplayName("End Game State")]
+        public Save.EndGameType EndGameState;
+
+        [UnrealFieldCategory("1. Information")]
+        [UnrealFieldDisplayName("Time Stamp")]
+        public Save.SaveTimeStamp TimeStamp;
+
+        [UnrealFieldCategory("3. Location")]
+        [UnrealFieldDisplayName("Position")]
+        public Save.Vector SaveLocation;
+
+        [UnrealFieldCategory("3. Location")]
+        [UnrealFieldDisplayName("Rotation")]
+        public Save.Rotator SaveRotation;
+
+        [UnrealFieldCategory("5. Other")]
+        [UnrealFieldDisplayName("Current Loading Tip")]
+        public int CurrentLoadingTip;
+
+        [UnrealFieldCategory("5. Other")]
+        [UnrealFieldDisplayName("Levels")]
+        public List<Save.Level> LevelRecords;
+
+        [UnrealFieldCategory("5. Other")]
+        [UnrealFieldDisplayName("Streaming")]
+        public List<Save.StreamingState> StreamingRecords;
+
+        [UnrealFieldCategory("5. Other")]
+        [UnrealFieldDisplayName("Kismet")]
+        public List<Save.KismetBool> KismetRecords;
+
+        [UnrealFieldCategory("5. Other")]
+        [UnrealFieldDisplayName("Doors")]
+        public List<Save.Door> DoorRecords;
+
+        [UnrealFieldCategory("5. Other")]
+        [UnrealFieldDisplayName("Pawns")]
+        public List<Guid> PawnRecords;
+
+        [UnrealFieldCategory("2. Squad")]
+        [UnrealFieldDisplayName("Player")]
+        public Save.Player PlayerRecord;
+
+        [UnrealFieldCategory("2. Squad")]
+        [UnrealFieldDisplayName("Henchmen")]
+        public List<Save.Henchman> HenchmanRecords;
+
+        [UnrealFieldCategory("4. Plot")]
+        [UnrealFieldDisplayName("ME2 Plot Table")]
+        public Save.PlotTable PlotRecord;
+
+        [UnrealFieldCategory("4. Plot")]
+        [UnrealFieldDisplayName("ME1 Plot Table")]
+        public Save.ME1PlotTable ME1PlotRecord;
+
+        [UnrealFieldCategory("4. Plot")]
+        [UnrealFieldDisplayName("Galaxy Map")]
+        public Save.GalaxyMap GalaxyMapRecord;
+
+        [UnrealFieldCategory("1. Information")]
+        [UnrealFieldDisplayName("Dependent DLC")]
+        public List<Save.DependentDLC> DependentDLC;
+
+        protected abstract void Serialize(IUnrealStream stream);
+
+        public abstract void Save(Stream output);
+
+        public static SaveFileBase Load(Stream input)
+        {
+            const Endian endian = Endian.Little;
+            const int MassEffect2LegendaryVersionSave = 30;
```

Comment:
I'd say this would be better as a const on the class itself, `public const LegendaryVersion = 30;`?

Reply:
done